### PR TITLE
Change database name for the context.

### PR DESCRIPTION
**Pull request by EminUZUN**
Fix for the issue:
https://github.com/starburstdata/dbt-trino/issues/228 Database name is not getting changed in each run. So this fix is getting desired database for catalog_name = '{{ database }}'

## Checklist

- [X] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [x] `README.md` updated and added information about my change
- [x] I have run `changie new` to [create a changelog entry](https://github.com/starburstdata/dbt-trino/blob/master/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by EminUZUN** [comment]
information_schema is implictly selected yes. but database is not setted in the session. so it sees "target" database only.
Whenever you change database in session. sources file in catalog.json.
You can see below results when you generate docs with debug option.

This is result when database is not explictly set:

```sql
with tables as (
            select
                table_catalog as "table_database",
                table_schema as "table_schema",
                table_name as "table_name",
                table_type as "table_type",
                null as "table_owner"

            from source_db.INFORMATION_SCHEMA.tables
            where
                table_schema != 'information_schema'
                and
                table_schema in ('public')

        ),

        columns as (

            select
                table_catalog as "table_database",
                table_schema as "table_schema",
                table_name as "table_name",

                column_name as "column_name",
                ordinal_position as "column_index",
                data_type as "column_type",
                comment as "column_comment"

            from source_db.INFORMATION_SCHEMA.columns
            where
                table_schema != 'information_schema'
                and
                table_schema in ('public')

        ),

        table_comment as (

            select
                catalog_name as "table_database",
                schema_name as "table_schema",
                table_name as "table_name",
                comment as "table_comment"

            from system.metadata.table_comments
            where
                catalog_name = 'target_db' --database name set wrong.
                and
                schema_name != 'information_schema'
                and
                schema_name in ('public')
        )

        select *
        from tables
        join columns using ("table_database", "table_schema", "table_name")
        join table_comment using ("table_database", "table_schema", "table_name")
        order by "column_index"
```

This is result when database implictly set:
```sql
  with tables as (

            select
                table_catalog as "table_database",
                table_schema as "table_schema",
                table_name as "table_name",
                table_type as "table_type",
                null as "table_owner"

            from source_db.INFORMATION_SCHEMA.tables
            where
                table_schema != 'information_schema'
                and
                table_schema in ('public')

        ),

        columns as (

            select
                table_catalog as "table_database",
                table_schema as "table_schema",
                table_name as "table_name",

                column_name as "column_name",
                ordinal_position as "column_index",
                data_type as "column_type",
                comment as "column_comment"

            from source_db.INFORMATION_SCHEMA.columns
            where
                table_schema != 'information_schema'
                and
                table_schema in ('public')

        ),

        table_comment as (

            select
                catalog_name as "table_database",
                schema_name as "table_schema",
                table_name as "table_name",
                comment as "table_comment"

            from system.metadata.table_comments
            where
                catalog_name = 'source_db' --database name set correctly.
                and
                schema_name != 'information_schema'
                and
                schema_name in ('public')
        )

        select *
        from tables
        join columns using ("table_database", "table_schema", "table_name")
        join table_comment using ("table_database", "table_schema", "table_name")
        order by "column_index"
```

**Comment by mdesmet** [comment]
What do you mean by 

> Whenever you change database in session

The database or catalog is set in the dbt profile and is static.

**Comment by mdesmet** [comment]
Never mind I found the relevant dbt-core code.

https://github.com/dbt-labs/dbt-core/blob/1250f23c445a733a58f1a5c4c186a6cbfd3106a3/core/dbt/adapters/base/impl.py#L1038-L1046

File: dbt/include/trino/macros/catalog.sql [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {% macro trino__get_catalog(information_schema, schemas) -%}
     {%- call statement('catalog', fetch_result=True) -%}
+        {% set database = information_schema.database %}
```
Comment (mdesmet) [inline dbt/include/trino/macros/catalog.sql:3]: Please inline this expression in the query itself

**Review by mdesmet — APPROVED** [review]

File: dbt/include/trino/macros/catalog.sql (hunk 1 above)
Comment (EminUZUN) [reply dbt/include/trino/macros/catalog.sql:3]: There is only one query in file. Where should I move it?

File: .changes/unreleased/Fixes-20230214-151316.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+kind: Fixes
+body: 'Set database name in catalog.sql for correctly filtering table_comments '
```
Comment (mdesmet) [inline .changes/unreleased/Fixes-20230214-151316.yaml:2]:
```suggestion
body: 'Include comments of sources or models from other catalogs in catalog.json artifact'
```

File: dbt/include/trino/macros/catalog.sql (hunk 1 above)
Comment (mdesmet) [reply dbt/include/trino/macros/catalog.sql:3]: Just do it in line 51 (or line 50 in original file) catalog_name = '{{ information_schema.database }}'

**Comment by mdesmet** [comment]
@EminUZUN : Can you also have a look at the tests?

**Pull request closed without merging**